rotkehlchen/db/drivers/gevent.py

```diff
@@ -131,39 +130,88 @@ def close(self) -> None:
         self._cursor.close()
 
 
-def progress_callback() -> int:
-    if __debug__:
-        logger.trace('Got in the progress callback')
-    gevent.sleep(0)
-    if __debug__:
-        logger.trace('Going out of the progress callback')
-    return 0
+class DBConnectionType(Enum):
+    USER = auto()
+    TRANSIENT = auto()
+    GLOBAL = auto()
+
+
+# This is a global connection map to be able to get the connection from inside the
+# progress handler. Having a global mapping and 3 different progress callbacks is
+# a sort of ugly hack. If anybody knows of a better way to make it work let's improve it.
+# With this approach we have named connections and a different progress callback per connection.
+CONNECTION_MAP: Dict[DBConnectionType, 'DBConnection'] = {}
+
+
+def _progress_callback(connection: Optional['DBConnection']) -> int:
```

Comment:
Why is the int type needed?

Reply:
Because this is the type of the progress callback 


https://docs.python.org/3/library/sqlite3.html#sqlite3.Connection.set_progress_handler